scripts/west_commands/runners/blackmagicprobe.py

```diff
@@ -5,11 +5,59 @@
 '''Runner for flashing with Black Magic Probe.'''
 # https://github.com/blacksphere/blackmagic/wiki
 
+import os
 import signal
+import sys
+import glob
 from pathlib import Path
 
 from runners.core import ZephyrBinaryRunner, RunnerCaps
 
+# Default path for linux, based on the project udev.rules file.
+DEFAULT_LINUX_BMP_PATH = '/dev/ttyBmpGdb'
+
+LINUX_SERIAL_GLOB = '/dev/ttyACM*'
+DARWIN_SERIAL_GLOB = '/dev/cu.usbmodem*'
+
+def BlackMagicProbeGDBSerial(port):
+    '''Guess the GDB port for the probe.
+
+    Return the port to use, in order of priority:
+        - the port specified manually
+        - the port in the BMP_GDB_SERIAL environment variable
+        - a guessed one depending on the host
+    '''
+    if port:
+        return port
+
+    if 'BMP_GDB_SERIAL' in os.environ:
+        return os.environ['BMP_GDB_SERIAL']
+
+    platform = sys.platform
+    if platform.startswith('linux'):
+        if os.path.exists(DEFAULT_LINUX_BMP_PATH):
+            return DEFAULT_LINUX_BMP_PATH
+
+        ports = glob.glob(LINUX_SERIAL_GLOB)
+        if not ports:
+            raise RuntimeError(
+                    f'cannot find any valid port matching {LINUX_SERIAL_GLOB}')
+        return sorted(ports)[0]
+
+    elif platform.startswith('darwin'):
+        ports = glob.glob(DARWIN_SERIAL_GLOB)
+        if not ports:
+            raise RuntimeError(
+                    f'cannot find any valid port matching {DARWIN_SERIAL_GLOB}')
+        return sorted(ports)[0]
+
+    elif platform.startswith('win32'):
+        return 'COM0'
```

Comment:
COM0 is not valid, starts at COM1, though you can get the current IDs from the registry somewhere but this is good enough for now

Reply:
Woops, bumped to COM1 thanks. Yeah I looked up how to guess it and the best I found was https://stackoverflow.com/questions/12090503/listing-available-com-ports-with-python, don't like the idea of actually opening the port to see if it's there though. Happy to leave the smartness here for someone who actually uses it.